dsa/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+#![no_std]
+#![forbid(missing_docs, unsafe_code)]
+#![deny(rust_2018_idioms)]
```

Comment:
We generally use `warn` rather than `deny`, and enforce code is warning-free in CI

```suggestion
#![forbid(unsafe_code)]
#![warn(missing_docs, rust_2018_idioms)]
```

(`unsafe_code` is the one exception in crates where we don't want any)